module/product-collection/src/Application/Controller/Api/Element/ElementChangeAction.php

```diff
@@ -0,0 +1,143 @@
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Swagger\Annotations as SWG;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\PropertyAccess\Exception\InvalidPropertyPathException;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(
+ *     name="ergonode_product_collection_element_change",
+ *     path="/collections/{collection}/elements/{product}",
+ *     methods={"PUT"},
+ *     requirements={
+ *     "collection"="[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}",
+ *      "product"="[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"
+ *     },
+ * )
+ */
+class ElementChangeAction
+{
+    /**
+     * @var MessageBusInterface
+     */
+    private $messageBus;
+
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @param MessageBusInterface  $messageBus
+     * @param FormFactoryInterface $formFactory
+     */
+    public function __construct(
+        MessageBusInterface $messageBus,
+        FormFactoryInterface $formFactory
+    ) {
+        $this->messageBus = $messageBus;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * @IsGranted("PRODUCT_COLLECTION_UPDATE")
+     *
+     * @SWG\Tag(name="Product Collection")
+     * * @SWG\Parameter(
+     *     name="collection",
+     *     in="path",
+     *     type="string",
+     *     required=true,
+     *     description="Collection Id",
+     * )
+     * @SWG\Parameter(
+     *     name="product",
+     *     in="path",
+     *     type="string",
+     *     required=true,
+     *     description="Product Id",
+     * )
+     * @SWG\Parameter(
+     *     name="language",
+     *     in="path",
+     *     type="string",
+     *     description="Language code",
+     *     default="EN"
+     * )
+     * @SWG\Parameter(
+     *     name="body",
+     *     in="body",
+     *     description="Update workflow",
+     *     required=true,
+     *     @SWG\Schema(ref="#/definitions/element_update")
+     * )
+     * @SWG\Response(
+     *     response=204,
+     *     description="Success"
+     * )
+     * @SWG\Response(
+     *     response=400,
+     *     description="Validation error",
+     *     @SWG\Schema(ref="#/definitions/validation_error_response")
+     * )
+     *
+     *
+     * @param ProductCollection $productCollection
+     * @param Request           $request
+     *
+     * @return Response
+     */
+    public function __invoke(ProductCollection $productCollection, Request $request): Response
+    {
+        try {
+            $model = new ElementChangeFormModel();
+            $form = $this->formFactory->create(ElementChangeForm::class, $model, ['method' => Request::METHOD_PUT]);
```

Comment:
Model ElementChangeFormMode
Form: ElementChangeForm
Command: UpdateProductCollectionElementCommand
url: /collections/{collection}/elements/{product}

I'm a little confused because I don't really know what's going on here.
Does it add an element or a product or a product is an element? 
I change or update ??? 
Please, organize the naming and take one convention. 
I guess you are adding the product to the Collection.